R/literal_coercion_linter.R

```diff
@@ -12,23 +12,31 @@
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
 literal_coercion_linter <- function() {
-  coercers <- xp_text_in_table(paste0(
-    "as.",
-    c("logical", "integer", "numeric", "double", "character")
-  ))
+  rlang_coercers <- xp_text_in_table(
+    c("lgl", "int", "dbl", "chr")
+  )
+  base_coercers <- xp_text_in_table(
+      paste0("as.", c("logical", "integer", "numeric", "double", "character"))
+  )
+
   # notes for clarification:
   #  - as.integer(1e6) is arguably easier to read than 1000000L
   #  - in x$"abc", the "abc" STR_CONST is at the top level, so exclude OP-DOLLAR
   #  - need condition against STR_CONST w/ EQ_SUB to skip quoted keyword arguments (see tests)
+  #  - for {rlang} coercers, both `int(1)` and `int(1, )` need to be linted
+  not_extraction_or_scientific <- "expr[2][
+    not(OP-DOLLAR)
+    and (
+      NUM_CONST[not(contains(translate(text(), 'E', 'e'), 'e'))]
+      or STR_CONST[not(following-sibling::*[1][self::EQ_SUB])]
+    )
+  ]"
   xpath <- glue::glue("//expr[
-    expr[1][SYMBOL_FUNCTION_CALL[ {coercers} ]]
-    and expr[2][
-      not(OP-DOLLAR)
-      and (
-        NUM_CONST[not(contains(translate(text(), 'E', 'e'), 'e'))]
-        or STR_CONST[not(following-sibling::*[1][self::EQ_SUB])]
-      )
-    ]
+    (
+      expr[1][SYMBOL_FUNCTION_CALL[ {base_coercers} ]] and {not_extraction_or_scientific}
+    ) or (
+      expr[1][SYMBOL_FUNCTION_CALL[ {rlang_coercers} ]] and {not_extraction_or_scientific} and count(expr) = 2
```

Comment:
IINM count(expr)=2 also applies to the base case, right? it's just kind of redundant since trailing ',' is not allowed. but we could combine the conditions here for conciseness, WDYT?

Reply:
That worked! Great idea. 

Done.